tests/basic/storage_pool_test.go

```diff
@@ -2053,3 +2209,1705 @@ var _ = Describe("{VolUpdateAddDrive}", func() {
+			expectedSize := poolToBeResized.TotalSize * 2 / units.GiB
+
+			log.FailOnError(err, "Failed to check if Journal enabled")
+
+			log.InfoD("Current Size of the pool %s is %d", selectedPool.Uuid, poolToBeResized.TotalSize/units.GiB)
+			err = Inst().V.ExpandPool(selectedPool.Uuid, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, expectedSize)
+			dash.VerifyFatal(err, nil, "Pool expansion init successful?")
+
+			resizeErr := waitForPoolToBeResized(expectedSize, selectedPool.Uuid, journalStatus)
+			dash.VerifyFatal(resizeErr, nil, fmt.Sprintf("Verify pool %s on node %s expansion using auto", selectedPool.Uuid, stNode.Name))
+
+		})
+		opts := make(map[string]bool)
+		opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+		ValidateAndDestroy(contexts, opts)
+
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		AfterEachTest(contexts, testrailID, runID)
+	})
+
+})
+
+var _ = Describe("{PoolExpandWhileIOAndPXRestart}", func() {
+	//step1: create volume repl node n1 and n2 and start IO
+	//step2: during I/O restart px on n1 and at the same time expand the pool on n2
+	//step3: after n1 is back operational validate that n2 pool size is the new size
+	//step4: read/validate I/O after expansion
+	var testrailID = 51445
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/51445
+	var runID int
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("PoolExpandWhileIOAndPXRestart", "Initiate pool expansion and restart px on n1 and at the same time expand the pool on n2 where vol repl exists", nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+	var contexts []*scheduler.Context
+
+	stepLog := "should get the volume with IOs, expand the pool by resize-disk and restart PX on one the repl node"
+
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("rsizerepl-%d", i))...)
+		}
+		ValidateApplications(contexts)
+
+		stNodes := node.GetStorageNodes()
+		if len(stNodes) == 0 {
+			dash.VerifyFatal(len(stNodes) > 0, true, "Storage nodes found?")
+		}
+		volSelected, err := getVolumeWithMinRepl(contexts, 2)
+		log.FailOnError(err, "error identifying volume")
+		appVol, err := Inst().V.InspectVolume(volSelected.ID)
+		log.FailOnError(err, fmt.Sprintf("err inspecting vol : %s", volSelected.ID))
+		replPools := appVol.ReplicaSets[0].PoolUuids
+		storageNode1, err := GetNodeWithGivenPoolID(replPools[0])
+		log.FailOnError(err, fmt.Sprintf("Failed to get pool using UUID %s", replPools[0]))
+		selectedPool := replPools[1]
+		storageNode2, err := GetNodeWithGivenPoolID(selectedPool)
+		log.FailOnError(err, fmt.Sprintf("Failed to get pool using UUID %s", selectedPool))
+
+		var poolToBeResized *api.StoragePool
+		poolToBeResized, err = GetStoragePoolByUUID(selectedPool)
+		log.FailOnError(err, "Failed to get pool using UUID ")
```

Comment:
Same here add the UUID that is used to get the StoragePool, would be good for debugging purposes
log.FailOnError(err, "Failed to get pool using UUID [%s]", selectedPool.Uuid)

Reply:
fixed